collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh

```diff
@@ -1,311 +1,528 @@
-done
 
-# K8s directory state
-echo "Collecting k8s directory state"
-mkdir -p $TMPDIR/k8s/directories
-if [ -d /opt/rke/etc/kubernetes/ssl ]; then
-  find /opt/rke/etc/kubernetes/ssl -type f -exec ls -la {} \; > $TMPDIR/k8s/directories/findoptrkeetckubernetesssl 2>&1
-elif [ -d /etc/kubernetes/ssl ]; then
-  find /etc/kubernetes/ssl -type f -exec ls -la {} \; > $TMPDIR/k8s/directories/findetckubernetesssl 2>&1
-fi
+  for RANCHERAGENT in $RANCHERAGENTS; do
+    docker inspect $RANCHERAGENT > $TMPDIR/rancher/containerinspect/agent-$RANCHERAGENT 2>&1
+    docker logs $SINCE_FLAG -t $RANCHERAGENT > $TMPDIR/rancher/containerlogs/agent-$RANCHERAGENT 2>&1
+  done
 
-# K8s certs
-echo "Collecting k8s certificates"
-mkdir -p $TMPDIR/k8s/certs
-if [ -d /opt/rke/etc/kubernetes/ssl ]; then
-  CERTS=$(find /opt/rke/etc/kubernetes/ssl -type f -name *.pem | grep -v "\-key\.pem$")
-  for CERT in $CERTS; do
-    openssl x509 -in $CERT -text -noout > $TMPDIR/k8s/certs/$(basename $CERT) 2>&1
+  # K8s Docker container logging
+  techo "Collecting k8s component logs"
+  mkdir -p $TMPDIR/k8s/containerlogs
+  mkdir -p $TMPDIR/k8s/containerinspect
+  for KUBE_CONTAINER in "${KUBE_CONTAINERS[@]}"; do
+    if [ "$(docker ps -a -q -f name=$KUBE_CONTAINER)" ]; then
+            docker inspect $KUBE_CONTAINER > $TMPDIR/k8s/containerinspect/$KUBE_CONTAINER 2>&1
+      docker logs $SINCE_FLAG -t $KUBE_CONTAINER > $TMPDIR/k8s/containerlogs/$KUBE_CONTAINER 2>&1
+    fi
   done
-  if [ -d /opt/rke/etc/kubernetes/.tmp ]; then
-    mkdir -p $TMPDIR/k8s/tmpcerts
-    TMPCERTS=$(find /opt/rke/etc/kubernetes/.tmp -type f -name *.pem | grep -v "\-key\.pem$")
-    for TMPCERT in $TMPCERTS; do
-      openssl x509 -in $TMPCERT -text -noout > $TMPDIR/k8s/tmpcerts/$(basename $TMPCERT) 2>&1
+
+  # System pods
+  techo "Collecting system pod logs"
+  mkdir -p $TMPDIR/k8s/podlogs
+  mkdir -p $TMPDIR/k8s/podinspect
+  for SYSTEM_NAMESPACE in "${SYSTEM_NAMESPACES[@]}"; do
+    CONTAINERS=$(docker ps -a --filter name=$SYSTEM_NAMESPACE --format "{{.Names}}")
+    for CONTAINER in $CONTAINERS; do
+      docker inspect $CONTAINER > $TMPDIR/k8s/podinspect/$CONTAINER 2>&1
+      docker logs $SINCE_FLAG -t $CONTAINER > $TMPDIR/k8s/podlogs/$CONTAINER 2>&1
     done
+  done
+
+  # Node and pod overview
+  mkdir -p $TMPDIR/k8s/kubectl
+  KUBECONFIG=/etc/kubernetes/ssl/kubecfg-kube-node.yaml
+  docker exec -it kubelet kubectl get nodes -o wide --kubeconfig=$KUBECONFIG > $TMPDIR/k8s/kubectl/nodes 2>&1
+  docker exec -it kubelet kubectl describe nodes --kubeconfig=$KUBECONFIG > $TMPDIR/k8s/kubectl/nodesdescribe 2>&1
+  docker exec -it kubelet kubectl get pods -o wide --all-namespaces --kubeconfig=$KUBECONFIG > $TMPDIR/k8s/kubectl/pods 2>&1
+  docker exec -it kubelet kubectl get svc -o wide --all-namespaces --kubeconfig=$KUBECONFIG > $TMPDIR/k8s/kubectl/services 2>&1
+  docker exec -it kubelet kubectl get endpoints -o wide --all-namespaces --kubeconfig=$KUBECONFIG > $TMPDIR/k8s/kubectl/endpoints 2>&1
+
```

Comment:
Nice-to-have a list of: deployments, daemonsets, statefulsets, ingress, configmaps

Output of kubectl top nodes and kubectl top pods could be nice too.

Reply:
and CRDs too.

Reply:
The configmaps would be possible, the others I'm afraid don't have permission from the kubelet.

Did you want to have a list of all configmaps, something like: `kubectl get configmaps --all-namespaces`?

Reply:
Yes, all configmaps. Can kubelet get list of namespaces? That might be nice to have too.

Reply:
Done